internal/fsx/fsx.go

```diff
@@ -0,0 +1,19 @@
+// Package fsx contains file system extension
+package fsx
+
+import (
+	"fmt"
+	"os"
+)
+
+func Open(name string) (*os.File, error) {
+	file, err := os.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	info, err := file.Stat()
+	if info.IsDir() {
+		return nil, fmt.Errorf("input file is a directory: " + name)
+	}
+	return file, err
```

Comment:
```suggestion
	return file, nil
```

Let us just return `nil` here, since we're morally returning a sum type.